ir.ts

```diff
@@ -28,7 +28,7 @@ export type Expr<A> =
   | {  a?: A, tag: "builtin1", name: string, arg: Value<A> }
   | {  a?: A, tag: "builtin2", name: string, left: Value<A>, right: Value<A>}
   | {  a?: A, tag: "call", name: string, arguments: Array<Value<A>> } 
-
+  | {  a?: A, tag: "call_indirect", index: number, arguments: Array<Value<A>> } 
```

Comment:
How can this be a constant number for `index`? Different subclasses will have their methods at different indices, right? It's fine if you focused on fields first (actually quite reasonable as the group working on closures also needs to define this, and getting the definition right will make merging easier). However, this seems like the wrong design and this has to be a `Value<A>`.

Reply:
Thanks for pointing this out! 

We have updated our definition of `call_indirect` so that it works well for both closures and inheritance. It looks like this now:

`{  a?: A, tag: "call_indirect", fn: Expr<A>, name: string, methodOffset: Value<A>, arguments: Array<Value<A>> }  `.